File: my-folder/problems/longest_subarray_of_1's_after_deleting_one_element/solution.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int l=0, r=0;
        const int n = nums.size();
        int z = 0;
        int ans = 0;
        while (r < n) {
            if (nums[r++] == 0) {
                z++;
            }
            while (z > 1) {
                if(nums[l] == 0) {
                    z--;
                }
                l++;
            }
            ans = std::max(ans, r-l-1);
        }
        return ans;
    }
};

File: my-folder/problems/longest_subarray_of_1's_after_deleting_one_element/solution.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        start = 0
        count = 0
        ans = 0
        for end in range(len(nums)):
            if nums[end] == 0:
                count += 1
            
            while count > 1:
                if nums[start] == 0:
                    count -= 1
                start += 1
            
            ans = max(ans, end-start)
        return ans